### PR TITLE
Add an ADR for codifying organization membership

### PR DESCRIPTION
**Pull request by accorvin**
## Description

I propose that we begin codifying (via a yaml file) membership in the OpenDataHub-io GitHub organization and using the Peribolos tool to automate applying that membership. This change adds an ADR outlining this decision.

## How Has This Been Tested?

N/A

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious). (N/A)
- [x] The developer has manually tested the changes and verified that the changes work (N/A)

### CODE REVIEW

File: architecture-decision-records/ODH-ADR-0006-organization-membership-automation.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+| Group                            | Key Contacts                            | Date        | Impacted? |
+| -------------------------------- | ---------------------------------------- | ---------- | --------- |
+| Architects Team                  | @opendatahub-io/architects               |            | y |
+| Documentation Team               | @opendatahub-io/documentation            |            | y |
+| Exploring Team                   | @opendatahub-io/exploring-team           |            | y |
+| Model Serving Team               | @opendatahub-io/model-serving            |            | y |
+| Training & Experimentation Team  | @opendatahub-io/training-experimentation |            | y |
+| Platform Team                    | @opendatahub-io/platform                 |            | y |
+
+This change impacts all of ODH. Rather than
```
Comment (anishasthana) [inline architecture-decision-records/ODH-ADR-0006-organization-membership-automation.md:103]: Is something missing here?

File: architecture-decision-records/ODH-ADR-0006-organization-membership-automation.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  to be periodically renewed.
+
+## Risks
+
+* The automation will need to be maintained.
+  * We expect the DevOps team to own this automation as part of their long term plans to automate
+    membership in the Red-Hat-Data-Services org as required for Konflux.
+* When we previously attempted to use Peribolos the automation was flaky and a constant point
+  of frustration. To mitigate this:
+    * We will use Peribolos only for managing organization membership, not teams or repository permissions.
```
Comment (anishasthana) [inline architecture-decision-records/ODH-ADR-0006-organization-membership-automation.md:86]: What would the process be like for creating a brand new team in the ODH org? Are `Org Membership Maintainers` capable of creating new teams? Or is anyone in the org allowed to create new teams.

File: architecture-decision-records/ODH-ADR-0006-organization-membership-automation.md (hunk 1 above)
Comment (accorvin) [reply architecture-decision-records/ODH-ADR-0006-organization-membership-automation.md:103]: Indeed. I had originally planned to not use the table then changed my mind. Will update.

File: architecture-decision-records/ODH-ADR-0006-organization-membership-automation.md (hunk 2 above)
Comment (accorvin) [reply architecture-decision-records/ODH-ADR-0006-organization-membership-automation.md:86]: Anyone in the org is allowed to create teams. The set of organization owners will periodically audit the teams to ensure that we're adhering to the hierarchy we want.

**Review by jkoehler-redhat — COMMENTED** [review]
What prevents someone adding and approving another owner?

**Comment by accorvin** [comment]
> What prevents someone adding and approving another owner?

Nothing really other than our review process, but nothing prevents that today either so I'm inclined to consider that out of scope for now.

**Comment by dtrifiro** [comment]
This? https://github.com/opendatahub-io/org-management it has been archived.

```
This repository is no longer used to manage organization membership. Reach out to another member of the OpenDataHub organization if you need to be added. If you do not know any members of the OpenDataHub organization, membership in the opendatahub-io organization can be made by opening an [issue](https://github.com/opendatahub-io/org-management/issues/new/choose).```

**Review by etirelli — APPROVED** [review]

**Pull request merged**